%\clearpage
%\addcontentsline{toc}{chapter}{Acronyms}
\section*{Acronyms}
Acronyms and their full forms used across the chapters in this thesis are listed down: 

\begin{multicols}{2}
\begin{description}[leftmargin=0pt]
\scriptsize{
        \item [AAE] Adversarial Autoencoder
        \item [ARI] Adjusted Rand Index
        %\item [ALWMMFC] Autoencoder Loss with Mixture Model for Clusters
        \item [AE] Autoencoder
        \item [ACC] Adrenocortical Cancer	
        \item [AEx] Adversarial Examples
    	\item [AUC] Area Under Receiver Operating Characteristic Curves
    	\item [BO] Bayesian Optimization
    	\item [BLCA] Bladder Urothelial Carcinoma
    	\item [BRCA] Breast Invasive Carcinoma
    	%\item[BACH] BreAst Cancer Histology images
    	\item [CNN] Convolutional Neural Network
        %\item[CC] Centroid-based Clustering
        %\item[CC1] Clustering Classification
        \item [CAH] Clustering Assignment Hardening
        \item [CHOL] Cholangiocarcinoma	
        \item [COAD] Colon Adenocarcinoma
        \item [CESC] Cervical and Endocervical Cancer
        \item [CDEC] Convolutional Deep Embedded Clustering
        %\item[CCNN] Clustering using CNN
        \item [CAE] Convolutional Autoencoder
        \item [ACC] Clustering Accuracy
        \item [CAM] Class Activation Maps
        \item [CN] Copy Numbers
        \item [CNA] Copy Number Aberration
        \item [CDSS] Clinical Decision Support System
        \item [DLBC] Diffuse Large B-cell Lymphoma	
        %\item[CXR] Chest X-ray
        %\item[CNN] Convolutional Neural Network
        \item [CLRP] Contrastive Layer-wise Relevance Propagation
        \item [CNV] Copy Number Variations
        \item [CAC] Cyclic Cosine Annealing
        \item [CNS] Copy Number Segment
        \item [DL] Deep Learning
        \item [DLx] Description Logic
        \item [DBN] Deep Belief Networks
        %\item[DEC] Deep Embedded Clustering
        %\item[DCN] Deep Clustering Network
        \item [DR] Dimensionality Reduction
        %\item[DC] Distribution-based Clustering
        %\item[DC1] Density-based Clustering
        %\item[DCC] Deep Continuous Clustering
        %\item[DBC] Discriminatively Boosted Clustering
        \item [DSAE] Denoising Stacked AE
        \item [DEN] Deep Embedding Network
        %\item[DEPICT] Deep Clustering with CAE embedding
        \item [DAC] Deep Adaptive Image Clustering 
        \item [DNN] Deep Neural Networks
        \item [DL] Deep Learning 
        \item [DT] Decision Trees
        \item [DenseNet] Dense Convolutional Network
        \item [DTD] Deep Taylor Decomposition
        \item [EM] Expectation Minimization
        \item [ER] Estrogen Receptor
        \item [ESCA] Esophageal Carcinoma
        \item [FFNN] Feed-forward Neural Networks
        \item [FCL] Fully-Connected Layer
        \item [FI] Feature Importance
        %\item[FCN] Fully Convolutional Neural Network
        \item [FM] Feature Maps
        \item [FNR] False Negative Rate
        \item [FPR] False Positive Rates
        \item [GE] Gene Expression
        \item [GBM] Glioblastoma Multiforme
        \item [GDC] Genomic Data Commons
        \item [GB] Guided Backpropagation
        \item [GAG] Globally Averaged Gradients 
        \item [GDPR] General Data Protection Regulation
        \item [GV] Genomic Variants
        \item [GAP] Global Averaged Pooling 
        \item [GMM] Gaussian Mixture Model
        \item [GAN] Generative Adversarial Networks
        \item [GLW] Greedy Layer-wise Training
        \item [GBT] Gradient Boosted Trees
        \item [GOFAI] Good Old-Fashioned Artificial Intelligence
        \item [GRC] Genome Reference Consortium
        \item [Grad-CAM] Gradient-guided  Class-activation Maps
        \item [HM] Heat Maps
        \item [HNSC] Head and Neck Squamous Cellcarcinoma
        \item [HER2] Human Epidermal Growth Factor Receptor 2
        \item [HC] Hierarchical Clustering
        %\item[HQ] High Quality
        \item [HGE] Histogram Equalization
        \item [HDO] Human Disease Ontology
        \item [ICU] Intensive Care Unit
        \item [IFS] Incremental Feature Selection
        \item [IEx] Isoform Expression
        %\item[IMSAL] Info Maximization and Self-augmentation Loss
        %\item [JULE] Joint Unsupervised Learning of Deep Representation for Images
        \item [KLD] Kullback-Leibler Divergence
        \item [KNN] K-nearest Neighbours 
        \item [KB] Knowledge Base
        \item [KIRC] Kidney Clear Cell-carcinoma
        \item [KICH] Kidney Chromophobe
        \item [KIRP] Kidney Papillary Cell-carcinoma
        \item [KG] Knowledge Graph
        \item [LAML] Acute Myeloid-leukemia
        \item [LPL] Locality Preserving Loss 
        \item [LP+GS] Locality Preserving and Group Sparsity
        %\item[LQ] Low QualitY
        \item [LSTM] Long Short-Term Memory
        \item [LR] Learning Rate
        \item [LGG] Brain Lower-grade Glioma
        \item [LIHC] Liver Hepatocellular Carcinoma
        \item [LUAD] Lung Adenocarcinoma	
        \item [LRP] Layer-wise Relevance Propagation
        \item [LUSC] Lung Squamous Cell-carcinoma
        \item [MESO] Mesothelioma	
        \item [ML] Machine Learning 
        \item [MGI] Mouse Genome Informatics
        \item [MAI] Mean Absolute Impact
        %\item[ML] Machine Learning
        \item [MLP] Multilayer Perceptron
        \item [MMMC] Max Margin for Mixture Components
        %\item[MQ] Medium Quality
        \item [NMI] Normalized Mutual Information
        \item [NE] Number of Epochs
        \item [NB] Naive Bayes
        \item [NS] Number of Cycles
        \item [NE] Number of Epochs
        \item [NGS] Next Generation Sequencing
        \item [OOD] Out-of-distribution
        \item [OOG] Ontology of Genes and Genomes
        \item [OECC] Outlier Exposure with Confidence Control
        \item [OV] Ovarian Serous Cystadenocarcinoma
        %\item[K-medoids] Partitioning Around Medoids
        \item [PAAD] Pancreatic Adenocarcinoma
        \item [PD] Partial Derivatives
        \item [PCA] Principal Component Analysis
        \item [PCPG] Pheochromocytoma and Paraganglioma
        \item [PCAt] PanCancerAtlas
        \item [PRAD] Prostate Adenocarcinoma
        \item [PM] Permutation Model
        \item [PMF] Perona-Malik Filter
    	\item [PM] Prediction Maximization
    	\item [PPV] Positive Predictive Value
    	\item [POTSF] Proto-oncogenes with Tumor-suppressor Function
    	\item [RBM] Restricted Boltzmann Machines
    	\item [PR] Progesterone Receptor
    	\item [ROC] Receiver Operating Characteristic
    	\item [ResNet] Residual Network 
    	\item [ReLU] Rectified Linear Unit
    	\item [READ] Rectum Adenocarcinoma
    	\item [RNN] Recurrent Neural Networks
    	\item [RS] Relevance Score
        \item [RL] Representation Learning
        \item [RL1] Reconstruction Loss
        \item [RF] Random Forest
        \item [SW] Semantic Web
        \item [SARC] Sarcoma
        \item [SKCM] Skin Cutaneous Melanoma
        \item [SVM] Support Vector Machines
    	\item [SM] Saliency Maps
    	\item [SCF] Self-consistent Field
    	\item [SHAP] SHapley Additive exPlanations
    	\item [SGD] Stochastic Gradient Descent
    	\item [SBRL] Scalable Bayesian Rule List
    	\item [SCPA] Softmax Class Posterior Averaging
        \item [SOM] Self-Organizing Maps
        \item [SNV] Single Nucleotide Variation
        \item [SNP] Single Nucleotide Polymorphism
        \item [SGD] Stochastic Gradient Descent
    	\item [SGLRP] Softmax-gradient LRP
    	\item [STAD] Stomach Adenocarcinoma
    	\item [TCGA] The Cancer Genome Atlas
    	\item [TL] Transfer Learning
    	\item [TGCT] Testicular Germcell Tumor
    	\item [TPR] True Positive Rate
    	\item [TF] Transcription Factors
    	\item [THCA] Thyroid Carcinoma
    	\item [THYM] Thymoma
    	\item [UCS] Uterine Carcinosarcoma
        \item [UDA] Unsupervised Data Augmentation
        \item [UCEC] Uterine Corpus Endometrioid Carcinoma
        \item [UVM] Uveal Melanoma
        \item [VAE] Variational Autoencoder
        \item [WCSS] Within-cluster Sum of Squares
        \item [XAI] eXplainable Artificial Intelligence
        }
\end{description}
\end{multicols}